components/accelerator/nvidia/error/xid/component.go

```diff
@@ -17,18 +17,73 @@ import (
 	"github.com/leptonai/gpud/log"
 )
 
+var xidComponent *component
+
 func New(ctx context.Context, cfg Config) components.Component {
 	cfg.Query.SetDefaultsIfNotSet()
 	setDefaultPoller(cfg)
 
 	cctx, ccancel := context.WithCancel(ctx)
 	getDefaultPoller().Start(cctx, cfg.Query, nvidia_component_error_xid_id.Name)
-
-	return &component{
+	// 1. create xid db
+	// 2. create local xid cache
+	// 2.1 warmup cache from db
+	// 3 watcher := dmesg_pkg.NewWatcher()
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				panic("context cancelled")
+			case rawDmesgLine, err := <-watcher.Watch():
+				// 1. parse xid
+				// 2. if not xid, skip
+				// 3. generate current event
+				//   3.1 fill in dmesg based xid
+				//   3.2 fill in nvml based xid (from current poller)
+				// 4. check local cache with timestamp/xid, if it is already in cache, skip
+				// 5. insert into db, cache
+			}
+		}
+	}()
+
+	go func() {
+		// 1. initialize base state
+		for {
+			// 1. invoke os component in past 3 days, get all reboot events
+			// 2. if succeed
+			// 2.1 update state with reboot events and cache
+			// 3. if failed
+			//NOTE: this might fail if the os component is not ready, and initial xid state might be wrong
+			break
+		}
```

Comment:
IMO we should warmup these reboot events when initializing, not in a separate goroutine. 
Do we need to read the reboot event regularly? Once rebooted, the world will be destroyed.